### PR TITLE
Refactor TransformGeometryImageFilter

### PR DESCRIPTION
**Pull request by blowekamp**
Im proves the conformance to expected ITK pipeline behaviors. 

Addresses bug related to:
 -  output information being update at the incorrect phase
 - composition pipeline execution escaping the GenerateData method
 - Not correctly updating region filter indicates it will

Derive class from InPlaceImageFilter, and set the default "RunInPlace" to true.

Made RigidTransform a pipeline object.

### CODE REVIEW

**Comment by dzenanz** [comment]
CircleCI has one failed test, due to:
```text
Name of Class = TransformGeometryImageFilter
Name of Superclass = InPlaceImageFilter
Class name is correct
Superclass name provided does not match object's Superclass::NameOfClass
```

File: Modules/Core/Transform/include/itkTransformGeometryImageFilter.hxx [hunk 1]
```diff
@@ -73,21 +74,12 @@ TransformGeometryImageFilter<TInputImage, TOutputImage>::GenerateData()
...
+  outputPtr->SetOrigin(newOriginPoint);
+  outputPtr->SetDirection(inverseRotation * this->GetInput()->GetDirection()); // NewDC = [R^-1][DC]
 
-  this->GraftOutput(m_OutputImage);
+  this->GraftOutput(outputPtr);
 }
 
-template <typename TInputImage, typename TOutputImage>
-void
-TransformGeometryImageFilter<TInputImage, TOutputImage>::PrintSelf(std::ostream & os, Indent indent) const
```
Comment (dzenanz) [inline Modules/Core/Transform/include/itkTransformGeometryImageFilter.hxx:84]: Why remove this?

File: Modules/Core/Transform/include/itkTransformGeometryImageFilter.h [hunk 2]
```diff
@@ -100,22 +100,22 @@ namespace itk
...
-class ITK_TEMPLATE_EXPORT TransformGeometryImageFilter : public ImageToImageFilter<TInputImage, TOutputImage>
+class ITK_TEMPLATE_EXPORT TransformGeometryImageFilter : public InPlaceImageFilter<TInputImage, TOutputImage>
 {
 public:
   ITK_DISALLOW_COPY_AND_MOVE(TransformGeometryImageFilter);
 
   /** Standard class type alias */
   using Self = TransformGeometryImageFilter;
-  using Superclass = ImageToImageFilter<TInputImage, TOutputImage>;
+  using Superclass = InPlaceImageFilter<TInputImage, TOutputImage>;
```
Comment (dzenanz) [inline Modules/Core/Transform/include/itkTransformGeometryImageFilter.h:110]: The test needs to be updated with new superclass name.

**Review by dzenanz — COMMENTED** [review]
Test needs fixing, otherwise looks good.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Transform/include/itkTransformGeometryImageFilter.hxx (hunk 1 above)
Comment (blowekamp) [reply Modules/Core/Transform/include/itkTransformGeometryImageFilter.hxx:84]: There are no longer an ivars to print :)

**Review by hjmjohnson — APPROVED** [review]
Thank you @blowekamp .  I am looking forward to using this filter from Python.  I currently need to run an external program saving temporary images and loading them back into the python script.

**Comment by blowekamp** [comment]
This class only handles versos transform. I'm adding another commit to reuse the `Transform::ApplyToImageMetaData` method to uniformly support linear transforms.

**Comment by blowekamp** [comment]
Current warning:
```
itkTransformGeometryImageFilter: warning(4): ITK type not wrapped, or currently not known: itk::DataObjectDecorator< itk::VersorRigid3DTransform< double > >
```
should be addressed in the next patch which chance the input to a decorated `itk::Transform`, which should already be wrapped..

**Review by dzenanz — APPROVED** [review]

**Comment by blowekamp** [comment]
@hjmjohnson Can you confirm/verify that the math in the original ResampleInPlace is equivalent to the Transform::ApplyToImageMetadata method?

**Review by dzenanz — APPROVED** [review]

**Comment by hjmjohnson** [comment]
> @hjmjohnson Can you confirm/verify that the math in the original ResampleInPlace is equivalent to the Transform::ApplyToImageMetadata method?

Not in the near future.  I am confident that @dzenanz carefully implemented it.

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**